### PR TITLE
add sender

### PR DESCRIPTION
**Pull request by ProDog**
add sender for `invoke`; add signers for `send`.

### CODE REVIEW

File: neo-cli/CLI/MainService.Contracts.cs [hunk 1]
```diff
@@ -42,13 +42,16 @@ private void OnDeployCommand(string filePath, string manifestPath = null)
...
+        /// <param name="signerAccounts">Signer's accounts</param>
         [ConsoleCommand("invoke", Category = "Contract Commands")]
-        private void OnInvokeCommand(UInt160 scriptHash, string operation, JArray contractParameters = null, UInt160[] signerAccounts = null)
+        private void OnInvokeCommand(UInt160 scriptHash, string operation, JArray contractParameters = null, UInt160 sender = null, UInt160[] signerAccounts = null)
         {
             Signer[] signers = Array.Empty<Signer>();
             if (signerAccounts != null && !NoWallet())
                 signers = CurrentWallet.GetAccounts().Where(p => !p.Lock && signerAccounts.Contains(p.ScriptHash)).Select(p => new Signer() { Account = p.ScriptHash, Scopes = WitnessScope.CalledByEntry }).ToArray();
+            if (sender != null)
+                signers = signers.Append(new Signer() { Account = sender, Scopes = WitnessScope.CalledByEntry }).ToArray();
```
Comment (shargon) [inline neo-cli/CLI/MainService.Contracts.cs:54]: This must be a `insert` in the first position. `Append` will put the sender at the end of the signers.

File: neo-cli/CLI/MainService.Contracts.cs [hunk 2]
```diff
@@ -42,13 +42,16 @@ private void OnDeployCommand(string filePath, string manifestPath = null)
...
+        /// <param name="signerAccounts">Signer's accounts</param>
         [ConsoleCommand("invoke", Category = "Contract Commands")]
-        private void OnInvokeCommand(UInt160 scriptHash, string operation, JArray contractParameters = null, UInt160[] signerAccounts = null)
+        private void OnInvokeCommand(UInt160 scriptHash, string operation, JArray contractParameters = null, UInt160 sender = null, UInt160[] signerAccounts = null)
         {
             Signer[] signers = Array.Empty<Signer>();
             if (signerAccounts != null && !NoWallet())
-                signers = CurrentWallet.GetAccounts().Where(p => !p.Lock && signerAccounts.Contains(p.ScriptHash)).Select(p => new Signer() { Account = p.ScriptHash, Scopes = WitnessScope.CalledByEntry }).ToArray();
+                signers = signerAccounts.Select(p => new Signer() { Account = p, Scopes = WitnessScope.CalledByEntry }).ToArray();
+            if (sender != null && !signerAccounts.Contains(sender))
```
Comment (shargon) [inline neo-cli/CLI/MainService.Contracts.cs:53]: If it is contained, it can be at the end of the list

Comment (ProDog) [reply neo-cli/CLI/MainService.Contracts.cs]: done

**Comment by ProDog** [comment]
This want to merge in https://github.com/neo-project/neo-node/pull/628. @shargon please have a look.

**Pull request merged**